src/trix/elements/trix_editor_element.js

```diff
@@ -238,6 +233,18 @@ export default class TrixEditorElement extends HTMLElement {
     this.editor?.loadHTML(this.defaultValue)
   }
 
+  get disabled() {
```

Comment:
The "disabled" bit is separated from the "element internals" change, right? I like both, but are they related?

Reply:
The integration with `ElementInternals` provides access to the [formDisabledCallback(disabled)](https://web.dev/articles/more-capable-form-controls#lifecycle_callbacks) lifecycle hook. Through that integration the element is given an opportunity to disable or re-enable its associated `<input>`. The `[disabled]` attribute is a [common attribute among form controls](https://developer.mozilla.org/en-US/docs/Learn/Forms/Basic_native_form_controls#common_attributes), and provides the server an opportunity to render the element with an initially disabled state. However, the `[disabled]` attribute is only one aspect of disabling an element. If it's rendered within a `<fieldset disabled>`, the `formDisabledCallback(disabled)` will fire (and the associated `<input>` will be disabled anyway if its a child of the `<fieldset>` whether `ElementInternals` is integrated or not). There is also a [:disabled](https://developer.mozilla.org/en-US/docs/Web/CSS/:disabled) CSS pseudo-selector that incorporates both `[disabled]` on the element itself _and_ `:disabled` of any ancestor `<fieldset>` element.

The `.disabled` property itself is an enhancement that's separate from the bare necessities. 

I've included it here because applications that want to introspect whether or not the element is disabled are unlikely to get it right when incorporating `[disabled]` on the element, `:disabled` on its `<fieldset>` ancestors (or their [HTMLFieldSetElement.disabled](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFieldSetElement/disabled) property).

Since the `<input>` element is built-in, it already accounts for all these permutations. The current `formDisabledCallback(disabled)` implementation already manages the `HTMLInputElement.disabled` property, so the `<input>` felt like a good delegate for the time being. If the `<trix-editor>` is ever changed to [managed its own value](https://web.dev/articles/more-capable-form-controls#setting-a-value) in a way that makes the `<input>` no longer necessary, this property could be changed to handle the rest of the logic.